### PR TITLE
Update deploy website script

### PR DESCRIPTION
**Pull request by CalvinKirs**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -30,4 +30,4 @@ jobs:
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
           publish_dir: build
-          publish_branch: asf-staging
\ No newline at end of file
+          publish_branch: asf-site
```
Comment (zhongjiajie) [inline .github/workflows/deploy.yml:33]: Add new line EOL? So as `.asf.yaml`

File: .asf.yaml [hunk 2]
```diff
@@ -14,6 +14,14 @@
...
+staging:
+  profile: ~
+  whoami: main
+  foo: trigger
+
+# asf-site branch will show up at https://seatunnel.apache.org
+publish:
+  whoami:  asf-site
 github:
   features:
```
Comment (zhongjiajie) [inline .asf.yaml:26]: I found out we enable issue in L28? Is expect behavior or add by mistake?

**Review by zhongjiajie — COMMENTED** [review]
Add NIT

File: .asf.yaml (hunk 2 above)
Comment (CalvinKirs) [reply .asf.yaml:26]: We need to specify the release branch

Comment (CalvinKirs) [reply .asf.yaml:26]: Enable branch-protection through the whoami parameter

**Pull request merged**